jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,659 @@
+     */
+    public void excludeMethod(String... methods)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _methods.exclude(methods);
+    }
+
+    /**
+     * Include {@link PathSpec}s in the conditions to be met
+     * @param paths The {@link PathSpec}s that are tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void include(PathSpec... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+
+        for (PathSpec p : paths)
+            ((PathSpecSet)_paths.getIncluded()).add(p);
+    }
+
+    /**
+     * Exclude {@link PathSpec}s in the conditions to be met
+     * @param paths The {@link PathSpec}s that are tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void exclude(PathSpec... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (PathSpec p : paths)
+            ((PathSpecSet)_paths.getExcluded()).add(p);
+    }
+
+    /**
+     * Include {@link PathSpec}s in the conditions to be met
+     * @param paths String representations of {@link PathSpec}s that are
+     *              tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void includePath(String... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _paths.include(paths);
+    }
+
+    /**
+     * Exclude {@link PathSpec} in the conditions to be met
+     * @param paths String representations of {@link PathSpec}s that are
+     *              tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void excludePath(String... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _paths.exclude(paths);
+    }
+
+    /**
+     * Include {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void include(InetAddressPattern... patterns)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (InetAddressPattern p : patterns)
+            _predicates.include(new InetAddressPredicate(p));
+    }
+
+    /**
+     * Include {@link InetAddressPattern}s in the conditions to be met
+     * @param patterns String representations of {@link InetAddressPattern}s that are
+     *              tested against the {@link ConnectionMetaData#getRemoteSocketAddress() getRemoteSocketAddress()} of
+     *              {@link Request#getConnectionMetaData()}.
+     */
+    public void includeInetAddress(String... patterns)
```

Comment:
This should just be an inner class InetAddressPredicate that is (eventually) composited together with the other Predicates for this `ConditionalHandler`

Reply:
OK, this one has to be done by predicate.   But I'm not understanding exactly what you mean by making an inner class?